tools/authors.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+# Copyright (c) 2017, Niklas Hauser
+# All rights reserved.
+#
+# The file is part of the xpcc library and is released under the 3-clause BSD
+# license. See the file `LICENSE` for the full license governing this code.
+# -----------------------------------------------------------------------------
+
+import subprocess
+
+# store the commits by author
+shortlog = {}
+
+# get the shortlog summary
+output = subprocess.Popen("git shortlog -sn", shell=True, stdout=subprocess.PIPE).stdout.read()
+# parse the shortlog
+for line in output.split("\n")[:-1]:
+    commits, author = line.split("\t")
+    shortlog[author] = int(commits)
+
+# some authors have aliases cos estupido
+author_alias = {
+    "Sascha Schade": [
+        "strongly-typed",
+        "Webb"
+    ],
+    "Kevin Läufer": [
+        "Laeufer",
+        "Kiwi",
+        "Kevin"
+    ],
+    "Daniel Krebs": [
+        "daniel"
+    ],
+    "Niclas Rohrer": [
+        "Niclas"
+    ],
+    "Martin Rosekeit": [
+        "martin",
+        "thundernail"
+    ],
+    "Georgi Grinshpun": [
+        "Georgi"
+    ],
+    "Thorsten Lajewski": [
+        "thorsten"
+    ],
+    "Hans Schily": [
+        "hans"
+    ]
+}
+
+# cleaned up dictionary
+commit_count = {}
```

Comment:
You could use [`commit_count = defaultdict(int)`](https://docs.python.org/2/library/collections.html#collections.defaultdict) here, remove the `add_commits` function below and just use `commit_count[author] += shortlog[slauthor]` to update the counts.